client/components/Draft.jsx

```diff
@@ -1,42 +1,86 @@
-import React, { useState } from "react";
+import React, { createRef, useState, useEffect } from "react";
+import { Head, DraftAddImage } from "../components";
+import { Row, Col } from "reactstrap";
+import { EditorState, convertToRaw, convertFromRaw } from "draft-js";
+import { Editor, createEditorState } from "medium-draft";
 
-import { addNewBlock } from "medium-draft";
+import { saveDescription, getDescription } from "../utils/apiWrapper";
+
+import "../public/styles/medium-draft.scss";
+import debounce from "lodash/debounce";
 
 export default function Draft(props) {
-  const [input, setInput] = useState(null);
-
-  const handleClick = () => input.click();
-
-  const handleChange = e => {
-    const { close, getEditorState, setEditorState } = props;
-    const file = e.target.files[0];
-    if (file.type.indexOf("image/") === 0) {
-      let reader = new FileReader();
-      reader.readAsDataURL(file);
-      reader.onload = () => {
-        const src = reader.result;
-        const newState = addNewBlock(getEditorState(), "atomic:image", { src });
-        setEditorState(newState);
-      };
+  const [editorState, setEditorState] = useState(createEditorState());
+  const [unsaved, setUnsaved] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [prevContent, setPrevContent] = useState(null);
+
+  const refsEditor = createRef();
+
+  const debounceSave = React.useCallback(
+    debounce(json => {
+      const { id, phaseName, stageName } = props;
+      saveDescription(id, phaseName, stageName, json);
+      setUnsaved(false);
+    }, 1000),
+    []
+  );
+
+  const handleChange = newState => {
+    if (!loading) {
+      setEditorState(newState);
+
+      const contentState = newState.getCurrentContent();
+      const json = JSON.stringify(convertToRaw(contentState));
+      if (json !== prevContent) {
+        debounceSave(json);
+        setUnsaved(true);
+        setPrevContent(json);
+      }
     }
-    close();
   };
 
+  useEffect(() => {
+    refsEditor.current.focus();
+    const { id, phaseName, stageName } = props;
+    getDescription(id, phaseName, stageName)
+      .then(data => {
+        const description = data.data.description;
+        setEditorState(
+          EditorState.createWithContent(convertFromRaw(JSON.parse(description)))
+        );
+        setLoading(false);
+        setPrevContent(description);
+      })
+      .catch(() => {
+        setLoading(false);
+      });
+  }, []);
```

Comment:
You can remove the array parameter entirely

Reply:
wait the two are different, empty array means on component mount whereas no array means on every rerender

Reply:
Ya i  need that in

Reply:
👍